### PR TITLE
fix(ui): LightWeightNoProjectMessage changed to not be so disruptive

### PR DESCRIPTION
**Pull request by davidenwang**
Previously setting the projects in the project store would trigger a DOM unmount, remount because of this lightweight component which would conditionally render its children or let the NoProjectMessage render its children. This was causing unnecessary refetching of data for the issues page which is being converted into the lightweight org context.

Now it will always render the NoProjectMessage but supply an additional prop to specify if the projects are loaded or not.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/lightWeightNoProjectMessage.tsx [hunk 1]
```diff
@@ -16,10 +16,13 @@ class LightWeightNoProjectMessage extends React.Component<Props> {
     if ('projects' in organization) {
       return <NoProjectMessage {...this.props} />;
     }
-    if (loadingProjects) {
-      return this.props.children;
-    }
-    return <NoProjectMessage {...this.props} projects={projects} />;
+    return (
+      <NoProjectMessage
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/lightWeightNoProjectMessage.tsx]: Can we also get rid of the above condition too?

File: tests/js/spec/views/projectsDashboard/noProjectMessage.spec.jsx [hunk 2]
```diff
@@ -55,4 +55,20 @@ describe('NoProjectMessage', function() {
...
   });
+
+  it('handles loading projects from props', function() {
+    const lightWeightOrg = TestStubs.Organization();
+    delete lightWeightOrg.projects;
+
+    const child = <div>child</div>;
+
+    const wrapper = shallow(
+      <NoProjectMessage projects={[]} loadingProjects organization={lightWeightOrg}>
```
Comment (billyvg) [inline tests/js/spec/views/projectsDashboard/noProjectMessage.spec.jsx]: Should we make sure lightweight version doesn't cause an unmount in a test?

Comment (davidenwang) [reply tests/js/spec/views/projectsDashboard/noProjectMessage.spec.jsx]: good idea, just added a test, let me know if that's the correct way to check for unmount/remount

Comment (billyvg) [reply tests/js/spec/views/projectsDashboard/noProjectMessage.spec.jsx]: Hmmm, it seems correct, but I would double check it by stashing changes to `lightWeightNoProjectMessage.tsx` and seeing if it fails?

Comment (davidenwang) [reply tests/js/spec/views/projectsDashboard/noProjectMessage.spec.jsx]: good call changed to use a class component with jest functions inside the component's lifecycle methods.

**Review by billyvg — APPROVED** [review]

**Pull request merged**